jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2282,45 +2282,83 @@ int toFlag() {
 			private final String className;
 			private final Lookup lookup;
 			private final int ClassOptFlags;
+
 			ClassDefiner(String name, byte[] template, Lookup lookupObj, int flags) {
 				className = name;
 				classBytes = template;
 				lookup = lookupObj;
 				ClassOptFlags = flags;
 			}
+
 			ClassDefiner(String name, byte[] template, Lookup lookupObj) {
 				className = name;
 				classBytes = template;
 				lookup = lookupObj;
 				ClassOptFlags = 0;
 			}
+
 			Class<?> defineClass(boolean initOption) {
 				Class<?> ret = AccessController.doPrivileged(new PrivilegedAction<Class<?>>() {
 					@Override
 					public Class<?> run() {
 						JavaLangAccess jlAccess = SharedSecrets.getJavaLangAccess();
 						Class<?> lookupClass = lookup.lookupClass();
-						return jlAccess.defineClass(lookupClass.getClassLoader(), lookupClass, className, classBytes, jlAccess.protectionDomain(lookupClass), initOption, ClassOptFlags, null);
+						return jlAccess.defineClass(lookupClass.getClassLoader(), lookupClass, className, classBytes,
+								jlAccess.protectionDomain(lookupClass), initOption, ClassOptFlags, null);
+					}
+				});
+				return ret;
+			}
+
+			Class<?> defineClass(boolean initOption, Object classData) {
+				Class<?> ret = AccessController.doPrivileged(new PrivilegedAction<Class<?>>() {
+					@Override
+					public Class<?> run() {
+						JavaLangAccess jlAccess = SharedSecrets.getJavaLangAccess();
+						Class<?> lookupClass = lookup.lookupClass();
+						return jlAccess.defineClass(lookupClass.getClassLoader(), lookupClass, className, classBytes,
+								jlAccess.protectionDomain(lookupClass), initOption, ClassOptFlags, classData);
```

Comment:
This duplicates the body of `defineClass`.  Can the ` Class<?> defineClass(boolean initOption) ` version forward to this one with a null `classData`?

Reply:
Yes. Addressed in the HEAD commit.